src/packages/solid/components/Button/Button.tsx

```diff
@@ -48,31 +46,56 @@ export interface ButtonStyleProps {
 const Button: Component<ButtonProps> = props => {
   return (
     <node
-      use:withPadding={styles.Container.padding}
+      use:withPadding={styles.Container?.padding}
       {...props}
-      style={styles.Container}
       animate
       forwardStates
+      tone={props.tone || styles.tone}
+      style={{
+        ...styles.Container,
+        ...styles.Container[props.tone || styles.tone],
+        ...props?.style?.Container
+      }}
     >
-      <View forwardStates style={styles.FlexContainer} width={props.width}>
+      <View
+        forwardStates
+        width={props.width}
+        tone={props.tone || styles.tone}
+        style={{
```

Comment:
Don't do this. It creates new objects for each component. Use style to set the default that won't change
This is better:
```jsx
 style={styles.FlexContainer}
          {
          ...styles.FlexContainer[props.tone || styles.tone],
          ...props?.style?.FlexContainer
        }

```

Reply:
this doesn't work because `...styles.FlexContainer[props.tone || styles.tone],` winds up overriding the `{...props}`. so if we try and set `color` from outside the component on a prop it gets overridden by the tone

Reply:
but if we combine the props spread with this, this works `{...{ ...styles.Container[props.tone || styles.tone], ...props }}` not sure why it's throwing errors unless i double spread like this though